src/main/java/com/powsybl/openloadflow/network/VoltageControl.java

```diff
@@ -59,4 +64,72 @@ public boolean isVoltageControlLocal() {
     public boolean isSharedControl() {
         return controllers.stream().flatMap(lfBus -> lfBus.getGenerators().stream()).filter(LfGenerator::hasVoltageControl).count() > 1;
     }
+
+    public void updateReactiveKeys() {
+        List<LfBus> controllerBuses = new ArrayList<>(controllers);
+
+        double[] reactiveKeys = createReactiveKeys(controllerBuses);
+
+        // no reactive dispatch on PQ buses, so we set the key to 0
+        for (int i = 0; i < controllerBuses.size(); i++) {
+            LfBus controllerBus = controllerBuses.get(i);
+            if (!controllerBus.isVoltageControllerEnabled()) {
+                reactiveKeys[i] = 0d;
+            }
+        }
+
+        // update bus reactive keys
+        double reactiveKeysSum = Arrays.stream(reactiveKeys).sum();
+        for (int i = 0; i < controllerBuses.size(); i++) {
```

Comment:
How do you know here that the sum is never equal to zero after updating the keys?

Reply:
Yes it might happen, when all generators of the control, go PQ. 
=> I added a check